--- libbimg-encode/src/image_encode.cpp.orig	2026-08-06 01:49:50
+++ libbimg-encode/src/image_encode.cpp	2026-08-16 22:04:20
@@ -6,11 +6,8 @@
 #include <bimg/encode.h>
 #include "bimg_p.h"
 
-#include <libsquish/squish.h>
-#include <etc1/etc1.h>
+#include <squish/squish.h>
 #include <etcpak/ProcessRGB.hpp>
-#include <nvtt/nvtt.h>
-#include <pvrtc/PvrTcEncoder.h>
 #include <edtaa3/edtaa3func.h>
 #include <astcenc.h>
 
@@ -19,7 +16,7 @@
 BX_PRAGMA_DIAGNOSTIC_IGNORED_MSVC(4702) // warning C4702: unreachable code
 BX_PRAGMA_DIAGNOSTIC_IGNORED_CLANG_GCC("-Wunused-parameter") // warning: unused parameter ‘alloc_context’ [-Wunused-parameter]
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
-#include <stb/stb_image_resize2.h>
+#include <stb_image_resize2.h>
 BX_PRAGMA_DIAGNOSTIC_POP();
 
 extern "C" {
@@ -136,7 +133,7 @@
 				break;
 
 			case TextureFormat::ETC1:
-				etc1_encode_image(src, _width, _height, 4, _width*4, dst);
+				BX_ERROR_SET(_err, BIMG_ERROR, "Unable to convert between input/output formats!");
 				break;
 
 			case TextureFormat::ETC2:
@@ -176,27 +173,8 @@
 				break;
 
 			case TextureFormat::PTC14:
-				{
-					using namespace Javelin;
-					RgbaBitmap bmp;
-					bmp.width  = _width;
-					bmp.height = _height;
-					bmp.data   = const_cast<uint8_t*>(src);
-					PvrTcEncoder::EncodeRgb4Bpp(dst, bmp);
-					bmp.data = NULL;
-				}
-				break;
-
 			case TextureFormat::PTC14A:
-				{
-					using namespace Javelin;
-					RgbaBitmap bmp;
-					bmp.width  = _width;
-					bmp.height = _height;
-					bmp.data   = const_cast<uint8_t*>(src);
-					PvrTcEncoder::EncodeRgba4Bpp(dst, bmp);
-					bmp.data = NULL;
-				}
+				BX_ERROR_SET(_err, BIMG_ERROR, "Unable to convert between input/output formats!");
 				break;
 
 			case TextureFormat::ASTC4x4:
@@ -243,7 +221,7 @@
 					}
 
 					astcenc_context* context;
-					status = astcenc_context_alloc(&config, 1, &context);
+					status = astcenc_context_alloc(&config, 1, &context, nullptr);
 
 					if (status != ASTCENC_SUCCESS)
 					{
@@ -327,11 +305,8 @@
 		switch (_dstFormat)
 		{
 		case TextureFormat::BC6H:
-			nvtt::compressBC6H(src, _width, _height, _width*16, _dst);
-			break;
-
 		case TextureFormat::BC7:
-			nvtt::compressBC7(src, _width, _height, _width*16, _dst);
+			BX_ERROR_SET(_err, BIMG_ERROR, "Unable to convert between input/output formats!");
 			break;
 
 		default:
